nixos/modules/virtualisation/virtualbox-image.nix

```diff
@@ -184,11 +207,11 @@ in {
           VBoxManage modifyvm "$vmName" \
             --memory ${toString cfg.memorySize} \
             ${lib.cli.toGNUCommandLineShell { } cfg.params}
-          VBoxManage storagectl "$vmName" --name SATA --add sata --portcount 4 --bootable on --hostiocache on
-          VBoxManage storageattach "$vmName" --storagectl SATA --port 0 --device 0 --type hdd \
+          VBoxManage storagectl "$vmName" ${lib.cli.toGNUCommandLineShell { } cfg.storageController}
+          VBoxManage storageattach "$vmName" --storagectl ${cfg.storageController.name} --port 0 --device 0 --type hdd \
```

Comment:
Your commit message indicates a logical change, but this diff seems to be a functional NOP to me. What am I missing?

Reply:
By default it should be a no-op but this makes it so that it isn't hard-coded and for VMware you can make it be SCSI, as it doesn't support SATA, see https://github.com/Graylog2/graylog2-images/issues/211

Reply:
Ah, I see.